File: java-api-examples/app/src/main/java/com/k2fsa/sherpaonnx/FeatureConfig.java
/*
 * // Copyright 2022-2023 by zhaoming
 */

package com.k2fsa.sherpaonnx;

public class FeatureConfig {
    final private int sampleRate;
    final private int featureDim;

    public FeatureConfig(int sampleRate, int featureDim) {
        this.sampleRate = sampleRate;
        this.featureDim = featureDim;
    }

    public int getSampleRate() {
        return sampleRate;
    }

    public int getFeatureDim() {
        return featureDim;
    }
}

File: sherpa-onnx/kotlin-api/FeatureConfig.kt
package com.k2fsa.sherpa.onnx

data class FeatureConfig(
    var sampleRate: Int = 16000,
    var featureDim: Int = 80,
    var dither: Float = 0.0f
)

fun getFeatureConfig(sampleRate: Int, featureDim: Int): FeatureConfig {
    return FeatureConfig(sampleRate = sampleRate, featureDim = featureDim)
}
